src/variants/lambda/common.mligo

```diff
@@ -0,0 +1,83 @@
+//    then calls that handlers "handler_check" procedure.
+let common_proposal_check (propose_params, ce : propose_params * contract_extra) : unit =
+  match unpack_proposal_metadata(propose_params.proposal_metadata) with
+    | Add_handler ahp -> if handler_exists(ahp.name, ce) then (failwith proposal_handler_exists : unit) else unit
+    | Remove_handler rhp -> if handler_exists_and_active(rhp, ce) then unit else (failwith proposal_handler_not_found : unit)
+    | Execute_handler ehp -> (match Big_map.find_opt ehp.handler_name ce.lambdas with
+          // Checks if the handler is active, and if it is, fetch its handler-check lambda
+          // and execute it with the input from this proposal.
+          | Some hinfo ->
+              if hinfo.is_active
+                then let
+                  hc_input = (ehp.packed_argument, ce.handler_storage)
+                  in hinfo.handler_check hc_input
+                else (failwith proposal_handler_not_found: unit)
+          | None -> (failwith proposal_handler_not_found: unit))
+
+let add_handler(ce, ahp : contract_extra * add_handler_param) : contract_extra =
+  if handler_exists(ahp.name, ce)
+    then
+     (failwith proposal_handler_exists: contract_extra)
+    else let
+      // Insert the new handler and mark it active.
+      handler_info = { code = ahp.code ; handler_check = ahp.handler_check; is_active = true }
+      in {ce with lambdas = Big_map.add ahp.name handler_info ce.lambdas }
+
+let disable_handler(ce, rhp : contract_extra * remove_handler_param) : contract_extra =
+  match Big_map.find_opt rhp ce.lambdas with
+    // If the handler exists, then mark it as inactive.
+    | Some hinfo -> {ce with lambdas = Big_map.update rhp (Some {hinfo with is_active = false}) ce.lambdas }
+    // If it does not, do nothing.
+    | None -> ce
+
+let execute_handler(proposer, frozen, ce, md, ehp : address * nat * contract_extra * proposal_metadata * execute_handler_param) : ((operation list) * contract_extra * (address option)) =
+  match Big_map.find_opt ehp.handler_name ce.lambdas with
+    | Some hinfo ->
+        // If handler is present, execute it. Note that we don't check for active status of the
+        // handler since we want inactive handlers to be executable as well, so that existing proposals
+        // that use the handler can proceed to completion. After execution, store the updated ja
+        // handler_storage, and return the emitted operations.
+        let
+          ph_input =
+            { packed_argument = ehp.packed_argument
+            ; handler_storage = ce.handler_storage
+            ; proposal_info = { from = proposer; frozen_token = frozen; proposal_metadata = md }
+            } in
+        let ph_out = hinfo.code ph_input
+        in (ph_out.operations, { ce with handler_storage = ph_out.handler_storage }, ph_out.guardian)
+    | None -> (failwith proposal_handler_not_found: ((operation list) * contract_extra * (address option)))
```

Comment:
If this happens doesn't that mean that there is a bug in the contract?

If so, I think it shouldn't be using the `proposal_handler_not_found` error IMO, as that's used for when a given parameter is incorrect.

Reply:
Have switched it to use `fail_decision_callback` error